src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -57,23 +63,20 @@ export function DiscussionThread(props: DiscussionThreadProps) {
   } = useGetRepliesInfinite(threadId, orderByDatePosted, limit)
   const replies = replyData?.pages.flatMap(page => page.results) ?? []
 
-  useEffect(() => {
-    const getMessageBody = async () => {
-      if (threadData) {
-        const messageUrl = await SynapseClient.getThreadMessageUrl(
-          threadData.messageKey,
-          accessToken,
-        )
-        const body = await getMessage(messageUrl.messageUrl)
-        setMessage(body)
-      }
-    }
-    getMessageBody()
-  }, [threadData, accessToken, threadId])
+  const onSave = (text: string, title?: string) => {
+    updateTitle({
+      title: title!,
+      threadId: threadId,
+    })
+    updateMessage({
+      messageMarkdown: text,
+      threadId: threadId,
+    })
+  }
```

Comment:
This logic should be moved to a ForumThreadEditor component. See other comments for details.